### ISSUE TITLE
Can't send SMS messages

### ISSUE THREAD

**Issue by dspacejs** [signalapp/Signal-Android#10340]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Signal can't send SMS messages 95% of the time. But can receive SMS messages, however.

I can send SMS messages with my default SMS app, but not Signal. I have also troubleshooted this with my provider. They changed settings on their back-end, which didn't fix it. Therefore the problem must be Signal.

### Steps to reproduce
- use my environment
- send SMS message
- it shouldn't send, staying as pending
- resend the same message to the same number using Graphene's default SMS app. It should work

**Actual result:** SMS is marked as pending, and never sends
**Expected result:** SMS should send without problems

### Device info
**Device:** Pixel 2 XL
**Android version:** 11. On the latest version of Graphene OS for this device
**Signal version:** 5.0.8

### Link to debug log
https://debuglogs.org/0a182e075a3656fe413631b3577f5ad052d6971a0c00adbfda5a430240ae94ea

**Comment by Genis-Sage** [comment]
hey @dspacejs 
according to your debug log:

> android.permission.READ_SMS: NO
> android.permission.RECEIVE_SMS: NO
> android.permission.SEND_SMS: NO
> android.permission.WRITE_SMS: YES

I looked at my phone Android One (Version 10), and I have the same permissions, when I disallow the SMS permission for Signal
if I give Signal the SMS permission, all 4 permissions are YES

and just for completeness:
right now, the issue could also be your GrapheneOS (either an [new behaviour from AOSP 11](https://developer.android.com/about/versions/11/privacy/permissions) or from some GrapheneOS specific function)

did you try out any other third party SMS app, like [Simple SMS](https://github.com/SimpleMobileTools/Simple-SMS-Messenger) ?

**Comment by dspacejs** [comment]
@Genis-Sage 

Thanks. I just checked in the settings, looks like the SMS permissions were off, despite the fact I set it as my default SMS app. I just turned it back on, but the same problem still exists.

It works with third-party SMS apps, I tried it on the default Graphene SMS app. Therefore the issue must be some issue with Signal, not Graphene. Also, I shouldn't have to get a new phone and updated OS just to send SMS messages with this app.

**Comment by Genis-Sage** [comment]
hey @dspacejs 
what other third-party SMS apps (aside from the GrapheneOS one) did you try out?

could you attach a new debug log with the SMS persmissions allowed?


also, what happens, if you try out the steps from other SMS sending issues?
#10233

#10270

**Comment by dspacejs** [comment]
@Genis-Sage 

> what other third-party SMS apps (aside from the GrapheneOS one) did you try out?

None. Didn't think it was necessary since it works with the stock SMS app, that clearly shows the problem is with Signal.

> could you attach a new debug log with the SMS persmissions allowed?

https://debuglogs.org/4433725b545b5755f50a3020f1a9c2e5c1b0cad371b2764aa73645eca6169b82

Didn't work. This time after a while it said "Send failed" instead of just being stuck on the loading spinner. So at least something has changed.

> also, what happens, if you try out the steps from other SMS sending issues?

No luck.

**Comment by baptiste0928** [comment]
I sometimes have the same problem on my phone (LineageOS 17.1 + MicroG), and I know someone who also had the same problem (Samsung Android 10, Stock ROM). SMS are marked as pending, and never sends. I need to kill the app and clear cache to allow the SMS to be sent.

Unfortunately I don't have any debug logs now, but I'll try to retrieve some if the problem happens again (it's quite frequent).

**Comment by greyson-signal** [comment]
@dspacejs  Looks like originally you didn't have `SEND_SMS` permissions:

```java
5.0.8 2020-12-27 09:35:31.806 ACDT W SmsSendJob: java.lang.SecurityException: Sending SMS message: uid 10111 does not have android.permission.SEND_SMS.
```


Then, after you manually added permissions, it looks like we're getting back a result of `101`, which corresponds to [`RESULT_RIL_SMS_SEND_FAIL_RETRY`](https://developer.android.com/reference/android/telephony/SmsManager#RESULT_RIL_SMS_SEND_FAIL_RETRY). That's a brand new response code in Android 11. 

```java
5.0.8 2020-12-29 15:15:39.568 ACDT I SmsSentJob: Got SMS callback: org.thoughtcrime.securesms.SendReceiveService.SENT_SMS_ACTION , 101
```

I've never seen that, not sure what to make of it at the moment. We could at least handle that as a retry, but it's unclear why it's failing to begin with.

**Comment by baptiste0928** [comment]
I've just encountered the problem again. It doesn't seem to have the same origin as @dspacejs, but I can't find anything particular in the logs.

I sent an SMS on 2021-01-06 around 12:10 pm, but it was still being sent. I had to force the application to stop and clear the cache so that it could be sent (after sending the debug logs).

Debug logs : https://debuglogs.org/5f0fe2132283581fd9185db1e182a1ee90b02d17542c5f3fc32369214e0f67e9

Should I create a new issue as the problem doesn't seem to be the same?
